scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/env python3
+"""
+This is a script to append an Archivematica UUID as
+an Internal-Sender-Identifier to existing born-digital bags.
+
+When run this will create a "target" folder in the same directory
+to work within. It will leave "some_bag_id.log" files for each
+bag it migrates.
+"""
+
+import datetime
+import hashlib
+import os
+import shutil
+import sys
+from uuid import UUID
+
+from elasticsearch import helpers
+from tqdm import tqdm
+
+from common import (
+    get_aws_resource,
+    get_aws_client,
+    get_storage_client,
+    get_secret,
+    get_elastic_client,
+)
+
+
+def generate_checksum(file_location):
+    sha256_hash = hashlib.sha256()
+
+    with open(file_location, "rb") as f:
+        for byte_block in iter(lambda: f.read(4096), b""):
+            sha256_hash.update(byte_block)
+
+        return sha256_hash.hexdigest()
+
+
+def compress_folder(folder, remove_folder=True):
+    archive_name = shutil.make_archive(folder, "gztar", folder)
+    if remove_folder:
+        shutil.rmtree(folder, ignore_errors=True)
+
+    return archive_name
+
+
+def filter_s3_objects(s3_client, bucket, prefix):
+    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+
+    if "Contents" in response:
+        return [content["Key"] for content in response["Contents"]]
+    else:
+        return []
+
+
+def load_space_separated_file(file_location, key_first=True):
```

Comment:
minor: took me a while to figure out what this function was for – FWIW, I think the term is "tag file".